json-rpc/src/runtime.rs

```diff
@@ -26,6 +28,54 @@ const LABEL_SUCCESS: &str = "success";
 const LABEL_BATCH: &str = "batch";
 const LABEL_SINGLE: &str = "single";
 
+#[derive(Schema)]
+struct HttpRequestLog<'a> {
+    #[schema(display)]
+    remote_addr: Option<std::net::SocketAddr>,
+    method: String,
+    path: String,
+    status: u16,
+    referer: Option<&'a str>,
+    user_agent: Option<&'a str>,
+    #[schema(debug)]
+    elapsed: std::time::Duration,
+}
+
+#[derive(Schema)]
+struct RpcRequestLog {
+    trace_id: u64,
+    request: Value,
+}
+
+#[derive(Schema)]
+struct RpcResponseLog<'a> {
+    trace_id: u64,
+    is_batch: bool,
+    response_error: bool,
+    response: &'a JsonRpcResponse,
+}
+
+#[macro_export]
+macro_rules! log_response {
+    ($trace_id: expr, $resp: expr, $is_batch: expr) => {
+        let mut level = Level::Debug;
+        if let Some(ref error) = $resp.error {
+            if is_internal_error(error) {
+                level = Level::Error
+            }
+        }
+        log!(
+            level,
+            RpcResponseLog {
+                trace_id: $trace_id,
+                is_batch: $is_batch,
+                response_error: $resp.error.is_some(),
+                response: $resp,
+            }
+        );
+    };
+}
```

Comment:
Thanks for making this change!

Reply:
Only nit here is that you may want to use fully qualified names to name the macro you're using (so you don't have to bring it in scope elsewhere). e.g `::libra_logger::log!(...)` instead of `log!(..)`